ibis/backends/duckdb/__init__.py

```diff
@@ -394,19 +389,10 @@ def do_connect(
             ("md:", "motherduck:", ":memory:")
         ):
             database = Path(database).absolute()
-
-        if temp_directory is None:
```

Comment:
I don't think this is a good idea to get rid of.

You're effectively making it so that in-memory connections no longer work on larger-than-memory data unless `temp_directory` is passed. That seems strictly worse than being inconsistent with `duckdb.connect`.

Reply:
In other words, your new test should pass without needing to provide the `temp_directory` argument.

Reply:
According to https://duckdb.org/docs/configuration/overview.html#global-configuration-options, it says the default temp_directory is "⟨database_name⟩.tmp or .tmp (in in-memory mode)", which makes me think that either way, duckdb is gonna make the temp dir

Reply:
oh, just saw max_temp_directory_size. yeah I will add a test that checks that if you
- do not pass temp dir
- set the memory limit low and do some computation
- then we expect `./.tmp/` to be created and have some content

Reply:
Oh, nice, it looks like it's now behaving as desired :)

In that case, can we remove the option entirely?

Reply:
Is that the only other behavior that we want to test for? Or is there another test case I should add that we want to pass?

Reply:
Actually we can't, because duckdb won't create intermediate directories

Reply:
Personally, I'd much rather have duckdb pooping files in the XDG-standard place (usually `~/.cache/$APPLICATION`), rather than in the current directory.

Reply:
If we're aiming for maximum transparency then we should literally just pass the behavior through and verify that out-of-core processing works.

Reply:
hmm, lol just a matter of taste I think, I would much rather temp files be in the current dir than somewhere else.

I think I would love to just wash our hands of it and pass through the kwargs transparently and not get bogged down. I just pushed some tests, but they are failing for some weird `duckdb.duckdb.TransactionException: TransactionContext Error: Failed to commit: failed to pin block of size 256.0 KiB (824.0 KiB/976.5 KiB used)` error that I don't want to have to work around.

Reply:
Like I'm not sure it should be obvious what to do. I could see arguments either way that we should 1. NOT do any magic creating dirs for people or 2. we definitely should create the needed intermediate dirs. Plenty of software does both, I think either would be sane.

Reply:
I think we should remove any special duckdb temp_directory handling and use consolidate on the upstream behavior.

Reply:
Effectively that should mean: remove all variable references to `temp_directory` and just let the argument pass through to the `duckdb.connect` call as part of `**kwargs`.

Reply:
Sounds good to me. Keep it simple. I'm gonna be out of internet for about a week, I can make those changes when I get back.

Reply:
I'll fix it up and wrap up the PR, thanks for getting it started!